pulsar-zookeeper-utils/src/main/java/com/yahoo/pulsar/zookeeper/ZooKeeperCache.java

```diff
@@ -289,28 +290,38 @@ public Boolean call() throws Exception {
 
             // Broker doesn't restart on global-zk session lost: so handling unexpected exception
             try {
-                this.zkSession.get().getData(path, watcher, (rc, path1, ctx, content, stat) -> {
+                // Put watch on the node no matter what
```

Comment:
Sure! This uses the capability of `ZooKeeper.exists` to put a watch on nonexistent nodes. The absence of this behavior caused an issue in which namespace bundle boundary updates were not being seen across all brokers. I will make a separate PR w/ just the cache change.